virtwho/configure.py

```diff
@@ -1,4 +1,119 @@
-"""
-Used to manage virt-who configure file
+"""Used to manage virt-who configure file
 Add/Delete/Update options for /etc/virt-who.conf, /etc/virt-who.d/xxx.conf
 """
+from virtwho.settings import Configure
+from virtwho.settings import config
+from virtwho.ssh import SSHConnect
+from virtwho.logger import getLogger
+
+logger = getLogger(__name__)
+
+
+class VirtwhoHypervisorConfig:
+    """Able to create and manage /etc/virt-who.d/xxx.conf file when
+    call this class"""
+    def __init__(self, mode, register):
+        """Create virt-who configuration file with basic options.
+         All data come from virtwho.ini file.
+        :param mode: The hypervisor mode.
+            (esx, xen, hyperv, rhevm, libvirt, kubevirt, local)
+        :param register: The subscription server. (rhsm, satellite)
+        """
+        self.remote_ssh = virtwho_ssh_connect(mode)
+        self.local_file = f'/root/{mode}.conf'
+        self.remote_file = f'/etc/virt-who.d/{mode}.conf'
+        self.section = f'virtwho-{mode}'
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+        if mode == 'local':
+            logger.info("Don't need to configure for local mode")
```

Comment:
Do we have cases which need to create a config file for libvirt local  in /etc/virt-who.d?

Reply:
Now no cases do that for local libvirt, but it's really a method to run local by config file. Because it only needs type= option, no need for any other configurations, so we don't config it.

```
[local]
type=libvirt

```

Reply:
Updated code to also create conf file for local mode.